packages/models/src/delegation/delegation.ts

```diff
@@ -62,11 +62,8 @@ export const Delegation = z.object({
   delegateId: TenantId,
   eserviceId: EServiceId,
   createdAt: z.coerce.date(),
-  submittedAt: z.coerce.date(),
-  approvedAt: z.coerce.date().optional(),
-  rejectedAt: z.coerce.date().optional(),
+  updatedAt: z.coerce.date().optional(),
```

Comment:
To make the model stricter I propose to make this mandatory and on creation set it to the same value used for `createdAt`
```suggestion
  updatedAt: z.coerce.date(),
```

Reply:
> To make the model stricter I propose to make this mandatory and on creation set it to the same value used for `createdAt`

I would keep it optional in order to keep the same behaviour with other domain items since `tenant`, `purpose` and `agreement` all have that property as optional